### ISSUE TITLE
Issue downloading multi package games

### ISSUE THREAD

**Issue by desmondarcher** [bucanero/pkgi-ps3#34]
Hi Bucanero,
I noticed there is an issue downloading games that are split into multiple package files. For example, pkgi is not able to download and install games if the URL for the game is a link to an XML file instead of a PKG file. For the games that link to XML files, the direct links to the PKG files that you need to install for that game are within that XML file. I can provide examples of necessary.

If it is not possible for this to be fixed that is fine. I understand that most games are not like this and are associated with a single PKG link and not an XML file that has multiple PKG links within it.

**Comment by bucanero** [comment]
yes, XML links are not currently supported, the app assumes that the URL is a .pkg file.

It could be added in a future version, but it's not a top priority right now.

**Comment by mohasi** [comment]
@desmondarcher Would be nice if you posted an example, in case this is worked on in the future.

**Comment by desmondarcher** [comment]
Sure, Yakuza 5 is an example.On Jul 15, 2020 6:51 PM, Mohammed Asif <notifications@github.com> wrote:
@desmondarcher Would be nice if you posted an example, in case this is worked on in the future.

**Comment by mohasi** [comment]
OK, so it looks like :

```xml
<hfs_manifest>
<file_name>UP0177-NPUB31658_00-YAKUZA5000000001_bg_1_4628521509dc8179ed5128cd117e3b9a40b2d58c.pkg</file_name>
<file_size>24228954192</file_size>
<number_of_split_files>6</number_of_split_files>
<pieces file_size="4687500000" hash_value="3fe2b40f00b96f75bbd22cf547697dc52f3a88c0" index="0" url="pkg link"/>
<pieces file_size="4687500000" hash_value="44a87748097dc227ee5156f305b22e78c7f3f28c" index="1" url="pkg link"/>
<pieces file_size="4687500000" hash_value="eca583be6d81114590d630247220f0b5fec40bbe" index="2" url="pkg link"/>
<pieces file_size="4687500000" hash_value="a691ae2681cbe41c1b7ee94e131c3ab21ee453e0" index="3" url="pkg link"/>
<pieces file_size="4687500000" hash_value="fad2dbbea1582b9bed82a0e01495cc49dd9ea6cf" index="4" url="pkg link"/>
<pieces file_size="791454192" hash_value="b7ebc00c173bfadcb7e7e64b7ccc577c6268235b" index="5" url="pkg link"/>
</hfs_manifest>
```

Are these meant to be installed separately or merged somehow then installed?

**Comment by desmondarcher** [comment]
They have to be merged for this particular game or the package installation will fail. You can merge these PKG files on a PC.On Jul 15, 2020 7:28 PM, Mohammed Asif <notifications@github.com> wrote:
OK, so it looks like :
<hfs_manifest>
<file_name>UP0177-NPUB31658_00-YAKUZA5000000001_bg_1_4628521509dc8179ed5128cd117e3b9a40b2d58c.pkg</file_name>
<file_size>24228954192</file_size>
<number_of_split_files>6</number_of_split_files>
<pieces file_size="4687500000" hash_value="3fe2b40f00b96f75bbd22cf547697dc52f3a88c0" index="0" url="pkg link"/>
<pieces file_size="4687500000" hash_value="44a87748097dc227ee5156f305b22e78c7f3f28c" index="1" url="pkg link"/>
<pieces file_size="4687500000" hash_value="eca583be6d81114590d630247220f0b5fec40bbe" index="2" url="pkg link"/>
<pieces file_size="4687500000" hash_value="a691ae2681cbe41c1b7ee94e131c3ab21ee453e0" index="3" url="pkg link"/>
<pieces file_size="4687500000" hash_value="fad2dbbea1582b9bed82a0e01495cc49dd9ea6cf" index="4" url="pkg link"/>
<pieces file_size="791454192" hash_value="b7ebc00c173bfadcb7e7e64b7ccc577c6268235b" index="5" url="pkg link"/>
</hfs_manifest>
Are these meant to be installed separately or merged somehow then installed?

**Comment by bucanero** [comment]
Note about XML downloads: 

A good test for someone with access to their purchases on the store, would be to queue one of these big xml downloads, and once it's queued then just pause it. Next, with either HEN or CFW, grab all the tasks files (`d0.pdb`, `d1.pdb`, `f0.pdb`, etc) from the `/dev_hdd0/vsh/game_pkg/` folder.

With those files I could probably get a clear view of how it is handled and maybe it can be replicated on pkgi.
